cluster-autoscaler/cloudprovider/cloud_provider.go

```diff
@@ -236,6 +236,23 @@ type NodeGroup interface {
 	GetOptions(defaults config.NodeGroupAutoscalingOptions) (*config.NodeGroupAutoscalingOptions, error)
 }
 
+// PolicyNodeGroup is a wrapper for a nodegroup that can have scaling policies
+type PolicyNodeGroup interface {
```

Comment:
@MaciekPytel @x13n WDYT about this kind of optional interface definition for cloud providers?

This would standardize what this looks like for all cloud providers but doesn't require any provider to use it at the cost of this type of runtime check:

`if policyNg, ok := ng.(PolicyNodeGroup); ok && policyNg.ScaleDownPolicy() == Hibernate {`